src/pages/blocks/centre.tsx

```diff
@@ -0,0 +1,327 @@
+  const handleFilterModalClose = () => setFilterModalOpen(false); // Function to close filter modal
+
+  const handleSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchInput(event.target.value);
+    filterCenters(event.target.value);
+  };
+
+  const filterCenters = (searchTerm: string) => {
+    setFilteredCenters(
+      centers?.filter((center) =>
+        center.toLowerCase().includes(searchTerm.toLowerCase())
+      )
+    );
+  };
+
+  const applyFilters = () => {
+    // Apply sorting and filtering based on selected options
+    let updatedCenters = centers;
+
+    // Apply sorting
+    if (sortOrder === 'asc') {
+      updatedCenters = updatedCenters?.sort();
+    } else if (sortOrder === 'desc') {
+      updatedCenters = updatedCenters?.sort().reverse();
+    }
+
+    // Apply center type filter
+    if (centerType === 'regular') {
+      updatedCenters = updatedCenters?.filter((center) =>
+        ['Bhivapur', 'Kamptee', 'Katol'].includes(center)
+      );
+    } else if (centerType === 'remote') {
+      updatedCenters = updatedCenters?.filter((center) =>
+        ['Kuhi', 'Mauda', 'Savner'].includes(center)
+      );
+    }
+
+    // Apply selected centers filter
+    if (selectedCenters.length > 0) {
+      updatedCenters = updatedCenters?.filter((center) =>
+        selectedCenters.includes(center)
+      );
+    }
+
+    setFilteredCenters(updatedCenters);
+  };
+
+  useEffect(() => {
+    const storedDistrict = localStorage.getItem('selectedBlockDistrict') || '';
+    setDistrict(storedDistrict);
+    
+    const storedBlock = localStorage.getItem('selectedBlock') || '';
+    setBlocks(storedBlock);
+  }, []);
+
+  useEffect(() => {
+    applyFilters();
+  }, [sortOrder, centerType, selectedCenters]);
+
+
+  return (
+    <Box minHeight="100vh">
+      <Header />
+      <Box sx={{ display: 'flex', justifyContent: 'center' }}>
+        <Box
+          width={'100%'}
+          sx={{ backgroundColor: 'white', padding: '1.2rem' }}
+        >
+          <Typography
+            textAlign={'left'}
+            fontSize={'22px'}
+            color={theme?.palette?.text?.primary}
+          >
+            {blocks}
+          </Typography>
+          <Typography
+            textAlign={'left'}
+            fontSize={'11px'}
+            mb={'1rem'}
+            fontWeight={'500'}
+            color={theme?.palette?.text?.primary}
+          >
+            {districts}, Maharashtra
+          </Typography>
+          <Box display={'flex'} alignItems={'center'} mb={2}>
+          <TextField
+          value={searchInput}
+          onChange={handleSearchChange}
+          placeholder={t('COMMON.SEARCH')}
+          variant="outlined"
+          size="medium"
+          sx={{
+            p: 2,
+                justifyContent: 'center',
+                height: '48px',
+                flexGrow: 1,
+                mr: 1,
+                backgroundColor: '#EDEDED',
```

Comment:
@Aar-if color should be in the custom theme

Reply:
Okay I'll check